modules/packages/Sort.chpl

```diff
@@ -283,64 +551,254 @@ proc QuickSort(Data: [?Dom] ?eltType, minlen=16, doublecheck=false, param revers
    :type reverse: `bool`
 
  */
-proc SelectionSort(Data: [?Dom], doublecheck=false, param reverse=false) where Dom.rank == 1 {
+proc SelectionSort(Data: [?Dom] ?eltType, doublecheck=false, param reverse=false) where Dom.rank == 1 {
   const lo = Dom.dim(1).low;
   const hi = Dom.dim(1).high;
   for i in lo..hi-1 {
     var (_, loc) = if reverse then maxloc reduce zip(Data[i..hi], {i..hi})
       else minloc reduce zip(Data[i..hi], {i..hi});
     Data(i) <=> Data(loc);
   }
+}
+
+
+/* Comparators */
+
+/* Default comparator used in sort functions.*/
+record DefaultComparator {
+
+  /*
+   Default compare method used in sort functions.
+
+   :arg a: Array element
+   :type a: `eltType`
+   :arg b: Array element
+   :type b: `eltType`
+   :returns: 1 if ``b < a``
+   :returns: 0 if ``a == b``
+   :returns: -1 if ``a < b``
+
+   */
+  proc compare(a:?t, b) {
+    if a < b { return -1; }
+    else if b < a { return 1; }
+    else return 0;
+  }
+}
 
-  if (doublecheck) then VerifySort(Data, "SelectionSort", reverse);
+/* Reverse comparator built from another comparator.*/
+record ReverseComparator {
```

Comment:
I like the symmetry of `ReverseComparator` and `DefaultComparator`. I considered `Reverse` and `Default` with `reverse` and `default` instances, but these would conflict with variables in the deprecated functions and would rob users of some excellent variable names.